### PR TITLE
chore(ct): change setProps to rerender

### PR DESCRIPTION
**Pull request by sand4rt**
partial fix for: https://github.com/microsoft/playwright/issues/15919

### CODE REVIEW

File: tests/components/ct-vue-vite/src/notation-jsx.spec.tsx [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
-  await expect(component).toContainText('Loading');
+test('renderer and keep the component instance intact', async ({ mount }) => {
+  const component = await mount<{ count: number }>(Counter, {
+    props: { 
+      count: 9001
+    }
+  });
+  await expect(component.locator('#rerender-count')).toContainText('9001')
+  
+  await component.rerender({ props: { count: 1337 } })
```
Comment (sand4rt) [inline tests/components/ct-vue-vite/src/notation-jsx.spec.tsx:18]: NOTE: JSX syntax is not supported yet. API Should be:

```typescript
 await component.rerender(<Counter count={1337} />)
```

File: packages/playwright-ct-vue/index.d.ts [hunk 2]
```diff
@@ -34,25 +34,22 @@ export type PlaywrightTestConfig = Omit<BasePlaywrightTestConfig, 'use'> & {
...
+  props?: Props,
+  slots?: Record<string, unknown>,
+  on?: Record<string, Function>,
+  hooksConfig?: any,
+}
+
+interface MountResult<Props = Record<string, unknown>> extends Locator {
   unmount(): Promise<void>;
-  setProps(props: Partial<Props>): Promise<void>
+  rerender(options: { props: Props }): Promise<void>
```
Comment (pavelfeldman) [inline packages/playwright-ct-vue/index.d.ts:46]: We are nearing the branch point, so we should probably call this `_rerender` for the time being, until we make it work for jsx + react + svelte.

**Review by pavelfeldman — APPROVED** [review]

**Comment by pooledge** [comment]
I'd love to have `rerender` for svelte very much, for example, in order to test some functional components e.g. accordion expandable etc.

**Comment by sand4rt** [comment]
> I'd love to have `rerender` for svelte very much, for example, in order to test some functional components e.g. accordion expandable etc.

Hi @pooledge, i've just created a PR for that. If you want to track it see: https://github.com/microsoft/playwright/pull/17940.

**Pull request merged**